From 9190ace9d1a226976d61ba648e295be4ebd035a3 Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Sun, 8 Dec 2019 12:03:42 -0800
Subject: [PATCH 19/51] lima: handle DRM_FORMAT_MOD_INVALID in
 resource_from_handle()

Assume that resource is tiled if we get DRM_FORMAT_MOD_INVALID
in resource_from_handle() and we don't have RO.

Fixes: 8c12f4e5f24f ("lima: enable tiling")
Reviewed-by: Qiang Yu <yuq825@gmail.com>
Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 src/gallium/drivers/lima/lima_resource.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gallium/drivers/lima/lima_resource.c b/src/gallium/drivers/lima/lima_resource.c
index b3d57e7cf01..be069f13999 100644
--- a/src/gallium/drivers/lima/lima_resource.c
+++ b/src/gallium/drivers/lima/lima_resource.c
@@ -329,9 +329,13 @@ lima_resource_from_handle(struct pipe_screen *pscreen,
    case DRM_FORMAT_MOD_ARM_16X16_BLOCK_U_INTERLEAVED:
       res->tiled = true;
       break;
+   case DRM_FORMAT_MOD_INVALID:
+      res->tiled = screen->ro == NULL;
+      break;
    default:
       fprintf(stderr, "Attempted to import unsupported modifier 0x%llx\n",
                   (long long)handle->modifier);
+      goto err_out;
    }
 
    return pres;
-- 
2.23.0

